common/workunit/workunit.cpp

```diff
@@ -14016,25 +14016,32 @@ bool executeGraphOnLingeringThor(IConstWorkUnit &workunit, const char *graphName
     return false;
 }
 
-static void setResources(StringBuffer &jobYaml, const IConstWorkUnit *workunit, const char *process)
+static void setResources(IPropertyTree *workerConfig, const IConstWorkUnit *workunit, const char *process)
 {
+    auto setResourcesItem = [&workerConfig](const char *category, const char *resourceName,  unsigned value, const char *units)
+    {
+        if (!value) return;
+        VStringBuffer xpath("spec/template/spec/containers/resources/%s@%s", category, resourceName);
```

Comment:
If end-users ever extend the deployment to have multiple containers, I am not sure if this will work.